AmplifyPlugins/Auth/AWSAuthPlugin/Dependency/AuthorizationProviderAdapter+SignedOutSession.swift

```diff
@@ -0,0 +1,69 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import AWSCore
+import AWSPluginsCore
+
+extension AuthorizationProviderAdapter {
+
+    func fetchSignedOutSession( _ completionHandler: @escaping SessionCompletionHandler) {
+        awsMobileClient.getAWSCredentials { awsCredentials, error in
+
+            guard error == nil else {
+                let authSession = AuthCognitoSignedOutSessionHelper.signedOutSession(withError: error!)
+                completionHandler(.success(authSession))
+                return
+            }
+
+            guard let credentials = awsCredentials else {
+                // This should not happen, throw an unknown error.
+                // Since we couldnot fetch aws credentials, we stop here to avoid sending partial information.
+                let error = AmplifyAuthError.unknown("""
+                    Could not fetch the credentials
```

Comment:
As elsewhere, let's spell out our expectations: why do we believe this shouldn't happen?